tests/fixtures/async-content/app/components/async-content.js

```diff
@@ -0,0 +1,20 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  fastboot: Ember.inject.service(),
+
+  init() {
+    this._super(...arguments);
+
+    let deferred = Ember.RSVP.defer();
+
+    Ember.run.later(() => {
+      if (!this.get('isDestroyed')) {
```

Comment:
this callback should not bet let run if the component has been destroyed. Instead on destroy, we should actually cancel this timer, and more importantly we should also cleanup/resolve the promise.